src/Juvix/Core/ErasedAnn/Conversion.hs

```diff
@@ -0,0 +1,60 @@
+module Juvix.Core.ErasedAnn.Conversion where
+
+import Data.List ((\\))
+import qualified Juvix.Core.Erased as Erased
+import Juvix.Core.ErasedAnn.Types
+import qualified Juvix.Core.Erasure.Types as E
+import qualified Juvix.Core.Types as Types
+import qualified Juvix.Core.Usage as Usage
+import Juvix.Library hiding (Type)
+
+free :: forall primTy primVal. E.Term primTy primVal -> [Symbol]
+free = Erased.free . E.eraseAnn
+
+convertTerm :: forall primTy primVal compErr m. (HasThrow "error" (Types.PipelineError primTy primVal compErr) m) => E.Term primTy primVal -> Usage.T -> m (AnnTerm primTy primVal)
+convertTerm term usage = do
+  let ty = E.getType term
+  ty' <- convertType ty
+  case term of
+    E.Var sym _ -> pure (Ann usage ty' (Var sym))
+    E.Prim p _ -> pure (Ann usage ty' (Prim p))
+    E.Let sym bind body (bindTy, _) -> do
+      -- Calculate captures.
+      let captures = Erased.free (Erased.Lam sym (E.eraseAnn body))
+      -- TODO: Is this the right usage?
+      bind <- convertTerm bind usage
+      body <- convertTerm body usage
+      bindTy <- convertType bindTy
+      -- TODO: Eventually add `let` to Michelson, probably, instead of this conversion.
+      let lamTy = Pi usage bindTy ty'
+          lam = Ann usage lamTy (LamM captures [sym] body)
+      pure (Ann usage ty' (AppM lam [bind]))
+    E.Lam sym body _ -> do
+      -- TODO: Is this the right usage?
+      body <- convertTerm body usage
+      case body of
+        -- Combine nested lambdas into multi-argument function.
+        Ann _ _ (LamM cap' arg' body') ->
+          pure (Ann usage ty' (LamM (cap' \\ [sym]) (sym : arg') body'))
+        _ ->
+          pure (Ann usage ty' (LamM (free term) [sym] body))
```

Comment:
Yes it would be best to just add `Let` to the Michelson backend instead of converting it. Would be rather simple to set this up in the InstructionsEff

Reply:
Sure, let's do this in a follow-up though - https://github.com/metastatedev/juvix/issues/436.